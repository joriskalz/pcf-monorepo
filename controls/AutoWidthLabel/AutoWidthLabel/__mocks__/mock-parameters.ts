/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
