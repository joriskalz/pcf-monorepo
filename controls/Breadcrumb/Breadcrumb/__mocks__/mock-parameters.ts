/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
