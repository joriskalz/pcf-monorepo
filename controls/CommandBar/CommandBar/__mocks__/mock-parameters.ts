/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
