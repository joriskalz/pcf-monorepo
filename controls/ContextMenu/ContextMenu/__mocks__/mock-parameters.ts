/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
