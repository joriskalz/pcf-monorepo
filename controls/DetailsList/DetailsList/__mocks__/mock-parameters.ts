/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
