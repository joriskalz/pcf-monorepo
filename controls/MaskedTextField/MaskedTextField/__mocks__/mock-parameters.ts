/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Value: new MockStringProperty(),
        Prefix: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        ErrorMessage: new MockStringProperty(),
        Mask: new MockStringProperty(),
        MaskFormat: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
    };
}
