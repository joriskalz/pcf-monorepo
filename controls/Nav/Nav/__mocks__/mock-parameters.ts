/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        CollapseByDefault: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
        SelectedKey: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
