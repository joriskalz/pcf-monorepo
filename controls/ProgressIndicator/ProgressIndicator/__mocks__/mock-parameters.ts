/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockWholeNumberProperty, MockDecimalNumberProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        Label: new MockStringProperty(),
        Description: new MockStringProperty(),
        TypeofIndicator: new MockEnumProperty(),
        PercentComplete: new MockWholeNumberProperty(),
        HideProgressBar: new MockTwoOptionsProperty(),
        BarHeight: new MockWholeNumberProperty()
    };
}
