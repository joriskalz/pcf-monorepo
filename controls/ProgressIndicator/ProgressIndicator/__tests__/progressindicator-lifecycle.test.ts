import { ProgressIndicator } from '..';
import { IInputs } from '../generated/ManifestTypes';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { getMockParameters } from '../__mocks__/mock-parameters';
import * as renderer from 'react-test-renderer';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

describe('ProgressIndicator', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.useFakeTimers();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
        // reset any jest timers
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if ((setTimeout as any).mock) {
            jest.runOnlyPendingTimers();
            jest.useRealTimers();
        }
    });

    it('theme', async () => {
        const { component, context } = createComponent();
        context.parameters.Theme.raw = JSON.stringify({
            palette: {
                themePrimary: '#test-primary',
            },
        });

        component.init(context);
        const spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
    });

    it('test type of Indicator', async() =>{
        const { component, context } = createComponent();
        component.init(context);
        context.parameters.TypeofIndicator.raw = 'Indeterminate Indicator';
        let spinnerComponent = renderer.create(component.updateView(context));
        expect(spinnerComponent.toJSON()).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new ProgressIndicator();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.Label.raw = 'Example Label';
    context.parameters.Description.raw = 'Example Description';
    context.parameters.AccessibilityLabel.raw = 'Progress Indicator';
    context.parameters.BarHeight.raw = 2;
    context.parameters.HideProgressBar.raw = false;
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, notifyOutputChanged, container, state };
}
