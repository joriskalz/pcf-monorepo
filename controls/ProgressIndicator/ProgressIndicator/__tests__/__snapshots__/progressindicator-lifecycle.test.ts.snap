// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator test type of Indicator 1`] = `
<div
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    className=
        ms-ProgressIndicator
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          width: -1px;
        }
  >
    <div
      className=
          ms-ProgressIndicator-itemName
          {
            color: #323130;
            line-height: 20px;
            overflow: hidden;
            padding-top: 4px;
            text-overflow: ellipsis;
            white-space: nowrap;
          }
      id="progress-indicator3-label"
    >
      Example Label
    </div>
    <div
      className=
          ms-ProgressIndicator-itemProgress
          {
            height: 2px;
            overflow: hidden;
            padding-bottom: 8px;
            padding-left: 0;
            padding-right: 0;
            padding-top: 8px;
            position: relative;
          }
    >
      <div
        className=
            ms-ProgressIndicator-progressTrack
            {
              background-color: #edebe9;
              height: 2px;
              position: absolute;
              width: 100%;
            }
            @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
              border-bottom: 1px solid WindowText;
            }
      />
      <div
        aria-describedby="progress-indicator3-description"
        aria-labelledby="progress-indicator3-label"
        className=
            ms-ProgressIndicator-progressBar
            {
              animation: keyframes 0%{left:-30%;}100%{left:100%;} 3s infinite;
              background-color: #0078d4;
              background: linear-gradient(to right, #edebe9 0%, #0078d4 50%, #edebe9 100%);
              height: 2px;
              min-width: 33%;
              position: absolute;
              transition: width .3s ease;
              width: 0px;
            }
            @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
              -ms-high-contrast-adjust: none;
              background-color: highlight;
              background: highlight;
              forced-color-adjust: none;
            }
        role="progressbar"
        style={
          Object {
            "transition": undefined,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className=
          ms-ProgressIndicator-itemDescription
          {
            color: #605e5c;
            font-size: 12px;
            line-height: 18px;
          }
      id="progress-indicator3-description"
    >
      Example Description
    </div>
  </div>
</div>
`;

exports[`ProgressIndicator theme 1`] = `
<div
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background: #ffffff;
        color: #323130;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
>
  <div
    className=
        ms-ProgressIndicator
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          width: -1px;
        }
  >
    <div
      className=
          ms-ProgressIndicator-itemName
          {
            color: #323130;
            line-height: 20px;
            overflow: hidden;
            padding-top: 4px;
            text-overflow: ellipsis;
            white-space: nowrap;
          }
      id="progress-indicator2-label"
    >
      Example Label
    </div>
    <div
      className=
          ms-ProgressIndicator-itemProgress
          {
            height: 2px;
            overflow: hidden;
            padding-bottom: 8px;
            padding-left: 0;
            padding-right: 0;
            padding-top: 8px;
            position: relative;
          }
    >
      <div
        className=
            ms-ProgressIndicator-progressTrack
            {
              background-color: #edebe9;
              height: 2px;
              position: absolute;
              width: 100%;
            }
            @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
              border-bottom: 1px solid WindowText;
            }
      />
      <div
        aria-describedby="progress-indicator2-description"
        aria-labelledby="progress-indicator2-label"
        className=
            ms-ProgressIndicator-progressBar
            {
              animation: keyframes 0%{left:-30%;}100%{left:100%;} 3s infinite;
              background-color: #test-primary;
              background: linear-gradient(to right, #edebe9 0%, #test-primary 50%, #edebe9 100%);
              height: 2px;
              min-width: 33%;
              position: absolute;
              transition: width .3s ease;
              width: 0px;
            }
            @media screen and (-ms-high-contrast: active), screen and (forced-colors: active){& {
              -ms-high-contrast-adjust: none;
              background-color: highlight;
              background: highlight;
              forced-color-adjust: none;
            }
        role="progressbar"
        style={
          Object {
            "transition": undefined,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className=
          ms-ProgressIndicator-itemDescription
          {
            color: #605e5c;
            font-size: 12px;
            line-height: 18px;
          }
      id="progress-indicator2-description"
    >
      Example Description
    </div>
  </div>
</div>
`;
